tests/unit/parsers/section-test.js

```diff
@@ -199,12 +199,14 @@ test("#parse handles multiple paragraphs in list item", assert => {
   parser = new SectionParser(builder);
   let sections = parser.parse(element);
 
-  assert.equal(sections.length, 1, "single list section");
+  assert.equal(sections.length, 2, '2 sections');
```

Comment:
This is the behaviour change which may need some thought. The previous/current behaviour smushes the text content of all elements in the `<li>` together which isn't ideal but it does preserve the overall "list" structure, the new behaviour gives preference to the nested content which preserves document structure such as headings and paragraphs but removes the list.

Reply:
We discussed this on Slack. When the DOM inside a `<li>` is trivial as in this test example you might want the smushing behavior, but as soon as there is arbitrarily complex content in the list item (many paragraphs, images, etc) considering this content as a candidate for top level sections is better behavior.